### ISSUE TITLE
az ad group: Group names with '&' in them get split at the point where the '&' appears

### ISSUE THREAD

**Issue by liamehenderson** [Azure/azure-cli#24209]
## Describe the bug

**Command Name**
`az ad group show --group "Hello_&_World" --query id --output tsv`

**Errors:**
```
No group matches the name of 'Hello_'

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- Have an AAD group with an '&' in the name
- `az ad group show --group {} --query id --output tsv`

## Expected Behavior

`az ad` command can handle group names that have "&" in them. Previously had to use a workaround with --% though with 2.40 this is apparently no longer needed https://github.com/Azure/azure-cli/blob/dev/doc/quoting-issues-with-powershell.md and was working fine until 2.41

## Environment Summary
```
Windows-10-10.0.19042-SP0
Python 3.10.5
Installer: MSI

azure-cli 2.41.0

Extensions:
azure-devops 0.17.0

Dependencies:
msal 1.20.0b1
azure-mgmt-resource 21.1.0b1
```

**Comment by yonzhan** [comment]
@jiasli for awareness

**Comment by liamehenderson** [comment]
Have put the --% workaround back into our scripts which seems to have solved it on 2.41

`az --% ad group show --group "Hello_&_World" --query id --output tsv`

**Comment by jiasli** [comment]
We reverted #23514 in #24015. `az` is now still `az.cmd`.
